pkg/notes/notes.go

```diff
@@ -212,6 +236,52 @@ func NoteTextFromString(s string) (string, error) {
 	return "", errors.New("no matches found when parsing note text from commit string")
 }
 
+func DocumentationFromString(s string) []*Documentation {
+	regex := regexp.MustCompile("(?s)```docs[\\r]?\\n(?P<text>.+)[\\r]?\\n```")
+	match := regex.FindStringSubmatch(s)
+
+	if len(match) < 1 {
+		// Nothing found, but we don't require it
+		return nil
+	}
+
+	result := []*Documentation{}
+	text := match[1]
+	text = stripStar(text)
+	text = stripDash(text)
+
+	scanner := bufio.NewScanner(strings.NewReader(text))
+	for scanner.Scan() {
+		const httpPrefix = "http"
+		s := strings.SplitN(scanner.Text(), httpPrefix, 2)
+		if len(s) != 2 {
+			continue
+		}
+		description := strings.TrimSpace(s[0])
+		urlString := httpPrefix + strings.TrimSpace(s[1])
+
+		// Validate the URL
+		parsedURL, err := url.Parse(urlString)
+		if err != nil {
+			continue
+		}
+
+		// Classify the URL type
+		docType := DocTypeExternal
+		if strings.Contains(parsedURL.Path, "/kubernetes/enhancements/") {
```

Comment:
I feel like we should always assume the DNS is GitHub. That's currently where the source of truth for KEPs are. Happy to debate this one though, I can see reasons not to.

Reply:
I agree and added a further check for that.